autoscaler/cluster.py

```diff
@@ -639,3 +648,7 @@ def maintain(self, cached_managed_nodes, running_insts_map,
                         else:
                             logger.info(
                                 '[Dry run] Would have terminated unmanaged %s', inst)
+
+    def _map_sets_to_values(self, set_map):
+        for key in set_map.keys():
+            set_map[key] = int(min(set_map[key]))
```

Comment:
The constraint that there only be one item in each set? No. 

Then again that would mean that someone passed something like `p2.xlarge=0,p2.xlarge=5,p2.8xlarge=7` in, requesting two different priorities for the same instance type.